.github/workflows/sync_on_push.yaml

```diff
@@ -0,0 +1,31 @@
+# Sync repos
+name: Sync repos
+
+on:
+  push:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+jobs:
+  sync-repo:
+    if: ${{ github.repository == 'zama-ai/concrete-ml' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+        with:
+          fetch-depth: 0
+      - name: Save repo
```

Comment:
what's the use of save repo ?

Reply:
My guess is that it's just a safeguard.
I copy-pasted it from @IceTDrinker 's workflow so maybe he can confirm.

Reply:
I preferred doing a snapshot of the repo as a safeguard as we are doing some force pushes in there